### ISSUE TITLE
Detect if marker is going to be clustered

### ISSUE THREAD

**Issue by gytiskv** [googlemaps/google-maps-ios-utils#84]
Is there any way to know when marker (cluster item) is going to be removed from the map and appeared in cluster?

**Comment by mountainvat** [comment]
By default, when zoom level changes on an integer boundary (1 to 2, 6 to 5, etc) a recluster will be triggered. The default clustering algorithms are deterministic but there's no cheap way to predict if an item will be grouped into a cluster or will be removed from a cluster unless a recluster is done (which you can call manually on the ClusterAlgorithm subject the a performance hit if there are thousands of items). There are 2 factors in deciding whether an item will be in a cluster or not:
- The output clusters of the clustering algorithm.
- The configuration of the renderer. By default if a cluster has less than 4 items, the items will be expanded out but this behavior can be changed.

What's the exact use case you are trying to achieve?

**Comment by gytiskv** [comment]
Thank you for your response. I've already changed renderer configuration of the min size to 3 items by changing the const.
The purpose of this to be done is that I need to know when a marker will be no longer visible on the map (clustered) to do some stuff. For instance, I need to hide some markers that are not items of any cluster and are added by code as simple markers (so they will never cluster automatically). Let's say they are the child of the parent (which is cluster item), and when the parent will hide, I also need to hide the child.
I found some way, but it really hits the performance, because it iterates all the clusters and markers, but the problem is, I still cannot know if the marker is clustered or not.
